### ISSUE TITLE
Transformation map 0 must have exactly one source file

### ISSUE THREAD

**Issue by toverux** [ampproject/remapping#153]
Hi,

I tried enabling vendor source maps on my Angular 13 project, and I get this error (relevant build log excerpt):

```
⚡ ng build home --configuration production
✔ Browser application bundle generation complete.

./node_modules/@apollo/client/cache/inmemory/fixPolyfills.js - Error: Module build failed (from ./node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js):
Error: Transformation map 0 must have exactly one source file.
Did you specify these with the most recent transformation maps first?
    at buildSourceMapTree (/cut/node_modules/@ampproject/remapping/dist/remapping.umd.js:729:23)
    at remapping (/cut/node_modules/@ampproject/remapping/dist/remapping.umd.js:831:23)
    at Object.result (/cut/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js:202:54)
    at Object.<anonymous> (/cut/node_modules/babel-loader/lib/index.js:221:43)
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/cut/node_modules/babel-loader/lib/index.js:3:103)
    at _next (/cut/node_modules/babel-loader/lib/index.js:5:194)
```

That file, `./node_modules/@apollo/client/cache/inmemory/fixPolyfills.js`, is empty, it only contains a map file reference:

```
⚡ cat node_modules/@apollo/client/cache/inmemory/fixPolyfills.js
//# sourceMappingURL=fixPolyfills.js.map% 
```

And that mapping file exists:

```
⚡ cat node_modules/@apollo/client/cache/inmemory/fixPolyfills.js.map
{"version":3,"file":"fixPolyfills.js","sourceRoot":"","sources":["../../../src/cache/inmemory/fixPolyfills.ts"],"names":[],"mappings":"","sourcesContent":["// Most JavaScript environments do not need the workarounds implemented in\n// fixPolyfills.native.ts, so importing fixPolyfills.ts merely imports\n// this empty module, adding nothing to bundle sizes or execution times.\n// When bundling for React Native, we substitute fixPolyfills.native.js\n// for fixPolyfills.js (see the \"react-native\" section of package.json),\n// to work around problems with Map and Set polyfills in older versions of\n// React Native (which should have been fixed in react-native@0.59.0):\n// https://github.com/apollographql/apollo-client/pull/5962\n"]}% 
```

I don't know if it occurs on other files, this is crashing the build, but I suspect there's something wrong with that file specifically.

Is that an issue that can be fixed in remapping, or should I post an issue somewhere else (eg. Apollo) ?

This is the version in use:

```
 yarn why @ampproject/remapping
=> Found "@ampproject/remapping@1.0.1"
info Reasons this module exists
   - "@angular-devkit#build-angular" depends on it
   - Hoisted from "@angular-devkit#build-angular#@ampproject#remapping"
```

Thanks in advance,

**Comment by jridgewell** [comment]
Looked into this today, and it's coming from angular-devkit's use of remapping: https://github.com/angular/angular-cli/blob/1a9da01b3a13f037b91c5502b6fa09055539bbe9/packages/angular_devkit/build_angular/src/babel/webpack-loader.ts#L261

This is telling us that we have 2 sourcemaps (a transformed and an original) via a shorthand form (the array argument), where we know the first sourcemap has only one original and the original is the next item in the array. The error is being thrown because the first sourcemap doesn't have exactly one source.

It's possible it has 0 sources, because there's no code. In which case, I can easily adjust the check to fix this. But if you're somehow generating a sourcemap with multiple sources, then this is a bug in angular-devkit and has to be fixed there.

Can you edit the error in `/home/morgan/Documents/MitM/mdc-core/node_modules/@ampproject/remapping/dist/remapping.umd.js:729:23` to tell us exactly how many sources are present? You should be able to add a `maps[i].sources.length` to the error message.

**Comment by toverux** [comment]
Hi @jridgewell, thanks for having taken a look!
So I did what you said and I can confirm that there is 0 sources in `maps[i].sources`.
And just adding a `;` in that file of Apollo made the build pass.

**Comment by jridgewell** [comment]
This is fixed in v1.0.2. You'll need to update your transitive dependencies, but it should now work with sourceless transforms like `@apollo/client/cache/inmemory/fixPolyfills.js`

**Comment by toverux** [comment]
It's working. Thank you very much!
